src/main/java/uk/gov/hmcts/reform/sendletter/data/LetterRepository.java

```diff
@@ -25,23 +28,47 @@
 @Repository
 public class LetterRepository {
 
-    private static final Logger log = LoggerFactory.getLogger(LetterRepository.class);
-
     private final NamedParameterJdbcTemplate jdbcTemplate;
 
     private final ObjectMapper objectMapper;
 
-    public LetterRepository(NamedParameterJdbcTemplate jdbcTemplate, ObjectMapper objectMapper) {
+    private final String resubmissionIntervalSqlCondition;
+
+    public LetterRepository(
+        NamedParameterJdbcTemplate jdbcTemplate,
+        ObjectMapper objectMapper,
+        @Value("${letter-resubmission-interval-in-hours}") int resubmissionInterval
+    ) {
         this.jdbcTemplate = jdbcTemplate;
         this.objectMapper = objectMapper;
+        this.resubmissionIntervalSqlCondition = String.format(
+            "+ %d * interval '1 hour' < CURRENT_TIMESTAMP", resubmissionInterval
+        );
     }
 
-    public void save(DbLetter letter, Instant creationTime, String messageId) throws JsonProcessingException {
-        jdbcTemplate.update(
-            "INSERT INTO letters "
-                + "(id, message_id, service, type, created_at, sent_to_print_at, printed_at, additional_data) "
-                + "VALUES "
-                + "(:id, :messageId, :service, :type, :createdAt, :sentToPrintAt, :printedAt, :additionalData::JSON)",
+    public SendLetterResponse save(DbLetter letter, Instant creationTime, String messageId)
+        throws JsonProcessingException {
+
+        return jdbcTemplate.execute(
+            "WITH new_letter AS ("
+                + "  INSERT INTO letters AS l"
+                + "  (id, message_id, service, type, created_at, sent_to_print_at, printed_at, additional_data)"
+                + "  VALUES"
+                + "  (:id, :messageId, :service, :type, :createdAt, :sentToPrintAt, :printedAt, :additionalData::JSON)"
+                + "  ON CONFLICT (message_id) DO UPDATE"
+                + "  SET submit_index = l.submit_index + 1, resubmitted_at = CURRENT_TIMESTAMP"
+                + "  WHERE l.message_id = :messageId"
+                + "    AND ("
+                + "      (l.resubmitted_at IS NULL AND l.created_at " + resubmissionIntervalSqlCondition + ") OR"
+                + "      (l.resubmitted_at IS NOT NULL AND l.resubmitted_at " + resubmissionIntervalSqlCondition + ")"
+                + "    )"
+                + "  RETURNING l.id, (SELECT 1 AS updated)"
+                + ") "
+                + "SELECT id, updated FROM new_letter "
+                + "UNION ALL "
+                + "SELECT id, 0 AS updated FROM letters "
+                + "WHERE message_id = :messageId "
+                + "LIMIT 1",
```

Comment:
can we simply search for a letter with the same `messageID` and put the logic in code instead, please?

Reply:
it is way more expensive to make 2 db calls. if desired action can be achieved with single query - i'd recommend to do so. especially for insert actions.

as one of the internet oligarchs said:
> best db transaction is the one which was never made

so - keep those calls minimum